src/backend/d3d12/TextureD3D12.cpp

```diff
@@ -122,9 +122,8 @@ namespace backend { namespace d3d12 {
         resourceDescriptor.Layout = D3D12_TEXTURE_LAYOUT_UNKNOWN;
         resourceDescriptor.Flags = D3D12ResourceFlags(GetAllowedUsage(), GetFormat());
 
-        mResource =
-            mDevice->GetResourceAllocator()->Allocate(D3D12_HEAP_TYPE_DEFAULT, resourceDescriptor,
-                                                      D3D12TextureUsage(GetUsage(), GetFormat()));
+        mResource = mDevice->GetResourceAllocator()->Allocate(
```

Comment:
Could you explain this change a bit?

Reply:
The change is that `D3D12TextureUsage(GetUsage(), GetFormat())` becomes `D3D12_RESOURCE_STATE_COMMON`. This is used by the resource allocator to pass as the `InitialResourceState` argument of [`ID3D12Device::CreateCommittedResource`](https://docs.microsoft.com/en-us/windows/desktop/api/d3d12/nf-d3d12-id3d12device-createcommittedresource).

Previously this would set the resource state to the one corresponding to the initial usage.Since we don't have initial usages anymore, we will always generate a resource state transition (until we optimize that) so we put a state that's always valid.

Reply:
Makes sense, thanks!